src/test/java/hudson/plugins/git/extensions/impl/CloneOptionHonorRefSpecTest.java

```diff
@@ -0,0 +1,75 @@
+package hudson.plugins.git.extensions.impl;
+
+import hudson.model.FreeStyleBuild;
+import hudson.model.FreeStyleProject;
+import hudson.model.Result;
+import org.jvnet.hudson.test.Issue;
+import hudson.plugins.git.AbstractGitTestCase;
+import hudson.plugins.git.BranchSpec;
+import hudson.plugins.git.GitSCM;
+import hudson.plugins.git.UserRemoteConfig;
+import org.eclipse.jgit.transport.RefSpec;
+import org.jenkinsci.plugins.gitclient.CloneCommand;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.Parameterized;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.List;
+
+@RunWith(Parameterized.class)
+public class CloneOptionHonorRefSpecTest extends AbstractGitTestCase {
+
+    private String refSpecVar;
+
+    public CloneOptionHonorRefSpecTest(String useRefSpecVariable) {
+        this.refSpecVar = useRefSpecVariable;
+    }
+
+    @Parameterized.Parameters(name = "{0}")
+    public static Collection permuteRefSpecVariable() {
+        List<Object[]> values = new ArrayList<>();
+        String[] allowed = {"${GIT_REVISION}", "${GIT_COMMIT}"};
+        for (String refSpecValue : allowed) {
+            Object[] combination = {refSpecValue};
+            values.add(combination);
+        }
+        return values;
+    }
+
+    /**
+     * This test confirms behavior of refspecs on initial clone with expanded variables.
+     * @throws Exception on error
+     */
+    @Test
+    @Issue("JENKINS-56063")
+    public void testRefSpecWithExpandedVariables() throws Exception {
+        List<UserRemoteConfig> repos = new ArrayList<>();
+        repos.add(new UserRemoteConfig(testRepo.gitDir.getAbsolutePath(), "origin", "+refs/heads/master:refs/remotes/origin/master" + refSpecVar, null));
```

Comment:
The branch name "master${GIT_REVISION}" or "master${GIT_COMMIT}" only works because of the happy accident that GIT_REVISION and GIT_COMMIT do not have a value when running in this test.  Using variables with an empty value does not test what we need to test.

I believe the definition of GIT_REVISION and GIT_COMMIT does not happen until after the checkout has been performed.  If that's the case, then I'm not sure how a user could use those values in their refspec.

I think the test needs to use environment variables that are defined by the Jenkins core, like JOB_BASE_NAME, NODE_NAME, and JENKINS_USER_NAME.

The test probably should also test a parameter value with a parameterized job definition.  The parameter might be USER_SELECTED_BRANCH_NAME or some other name that clearly shows it is provided by the user.